### PR TITLE
Containers

### PR DESCRIPTION
**Pull request by ivanmalagon**
## WIP

Implements #40

### CODE REVIEW

File: packages/styles/src/core/containers/_containers.scss [hunk 1]
```diff
@@ -0,0 +1,92 @@
+.as-sidebar {
```
Comment (IagoLast) [inline packages/styles/src/core/containers/_containers.scss]: I this styles should be defined in sidebar

File: packages/styles/src/core/containers/_containers.scss [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    .as-box {
+      flex-basis: auto;
+      flex-grow: 1;
+      flex-shrink: 0;
+      overflow: auto;
+    }
+  }
+}
+
+.as-panel__element {
```
Comment (IagoLast) [inline packages/styles/src/core/containers/_containers.scss]: this should be defined in panels.scss

File: packages/styles/src/core/containers/test/containers.spec.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+function image (filename) {
+  return path.resolve(__dirname, `img/${filename}`);
+}
+
+function html (filename) {
+  const filePath = path.resolve(__dirname, `${filename}`);
+  return `file://${filePath}`;
+}
+
+function spec (reference, screenshot, url, mobile) {
```
Comment (IagoLast) [inline packages/styles/src/core/containers/test/containers.spec.js]: Nice idea!  We could refactor all our tests and create some kind of helper so we write less boilerplate code

File: packages/styles/src/core/layout/panels/_panels.scss [hunk 4]
```diff
@@ -72,12 +75,15 @@
   }
 }
 
+$bottom-bar-max-height: 200px;
```
Comment (IagoLast) [inline packages/styles/src/core/layout/panels/_panels.scss]: `variables/_sizes.scss` :trollface:

**Comment by ivanmalagon** [comment]
@IagoLast CR suggestions done

**Comment by ivanmalagon** [comment]
Documentation is pending.

File: packages/styles/src/core/containers/_containers.scss [hunk 5]
```diff
@@ -0,0 +1,11 @@
+.as-container--primary {
```
[the remark this answers is not in the document]
Comment (IagoLast) [reply packages/styles/src/core/containers/_containers.scss]: - Pros
  - Less code
  - Less verbose for the user (single attribute instead a class)
- Cons
  - This breaks HTML validation (we could use a `data-*` instead)
  - Goes against BEM

Considerations:

- Probably we could refactor the Styles to get a valid markup without repeating code, I'm going to take a look.
- If we all like the `attribute` approach, we should update all our styles and use attr-modifiers instead BEM modifiers, IMO it's very important to be consistent across all elements/components

[the remark this answers is not in the document]
Comment (ivanmalagon) [reply packages/styles/src/core/containers/_containers.scss]: Could you @jesusbotella open an issue with this debate?

I think it's not a matter of this PR and first, we need to have a consensus and second, we'll need to refactor after the deccission is taken.

File: packages/styles/src/core/layout/panels/_panels.scss [hunk 6]
```diff
@@ -7,6 +7,7 @@
   top: 0;
   width: 100%;
   height: 100%;
+  overflow-y: auto;
```
[the remark this answers is not in the document]
Comment (ivanmalagon) [reply packages/styles/src/core/layout/panels/_panels.scss]: Moved to `--visible` where it makes sense.

File: packages/styles/src/core/layout/panels/_panels.scss [hunk 7]
```diff
@@ -19,15 +20,25 @@
...
+
+  &__element {
+    .as-box {
+      @include margin(10 10 0 10);
+    }
+  }
+
   @media all and (min-width: $tablet-breakpoint) {
     display: flex;
     position: absolute;
```
[the remark this answers is not in the document]
Comment (ivanmalagon) [reply packages/styles/src/core/layout/panels/_panels.scss]: No. `.as-panels` is the whole map viewport. `as-panel` is each one of the possible positions in that container.

File: packages/styles/src/core/containers/_containers.scss (hunk 5 above)
[the remark this answers is not in the document]
Comment (jesusbotella) [reply packages/styles/src/core/containers/_containers.scss]: https://github.com/CartoDB/airship/issues/127

File: packages/styles/src/core/layout/panels/_panels.scss (hunk 7 above)
[the remark this answers is not in the document]
Comment (jesusbotella) [reply packages/styles/src/core/layout/panels/_panels.scss]: Oh, that's right. I was wrong :(

**Pull request merged**